mlir/include/mlir/Support/Timing.h

```diff
@@ -351,6 +351,15 @@ class DefaultTimingManager : public TimingManager {
     Tree,
   };
 
+  /// The different output formats for printing the timers.
+  enum class OutputFormat {
```

Comment:
This enum seems like a poor basis for extensibility. It seems to me a more powerful API would be to directly expose the `OutputStrategy` abstract class and allow users to inject their own `std::unique_ptr<OutputStrategy>` implementation. We can ship Text and Json as "batteries" for it.

Reply:
Ping.